src/test/java/com/rultor/profiles/ProfilesTest.java

```diff
@@ -0,0 +1,208 @@
+ * with the distribution. 3) Neither the name of the rultor.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.rultor.profiles;
+
+import com.jcabi.xml.XMLDocument;
+import com.rultor.spi.Profile;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Tests for ${@link Profiles}.
+ * @author Maksimenko Vladimir (xupypr@xupypr.com)
+ * @version $Id$
+ * @since 1.62
+ */
+public class ProfilesTest {
+
+    /**
+     * Profile template.
+     */
+    private static final String PROFILE = StringUtils.join(
+        "<p><entry key='merge'>",
+        "<entry key='architect'><item>%s</item></entry>",
+        "<entry key='commander'><item>%s</item></entry>",
+        "<entry key='script'><item>%s</item></entry>",
+        "</entry></p>"
+    );
+
+    /**
+     * Template for exception message.
+     */
+    private static final String MESSAGE =
+        "You cannot change `%s` section for security reasons";
+
+    /**
+     * Profile validation fails if architect list was changed.
+     * @throws Exception In case of error.
+     */
+    @Test
+    public final void validationFailsOnArchitectsMismatch() throws Exception {
+        final String commander = "Yegor1024";
+        final Profile master = new Profile.Fixed(
+            new XMLDocument(
+                String.format(
+                    ProfilesTest.PROFILE,
+                    "Yegor64",
+                    commander,
+                    "do_something"
+                )
+            )
+        );
+        final Profile merged = new Profile.Fixed(
+            new XMLDocument(
+                String.format(
+                    ProfilesTest.PROFILE,
+                    "Bobby",
+                    commander,
+                    "do_another"
+                )
+            )
+        );
+        try {
+            new Profiles().validated(master, merged);
+            MatcherAssert.assertThat(
```

Comment:
@xupyprmv wouldn't it be better to use `Assert.fail` instead of assertion on `false`?